### PR TITLE
ddl: fix ddl modify column from null to not null bug: check null value again before change to no null (#10948)

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
cherry-pick https://github.com/pingcap/tidb/pull/10948

### CODE REVIEW

**Comment by bb7133** [comment]
How about cherry-picking https://github.com/pingcap/tidb/pull/13223 together?

File: ddl/db_test.go [hunk 1]
```diff
@@ -2675,50 +2675,73 @@ LOOP:
...
+	s.tk.MustExec("delete from t1")
+	hook.OnJobUpdatedExported = func(job *model.Job) {
+		if tbl.Meta().ID != job.TableID {
+			return
+		}
+		if job.State != model.JobStateRunning {
+			return
+		}
+		if times == 0 {
+			tk2.MustExec("insert into t1 values ();")
```
Comment (bb7133) [inline ddl/db_test.go]: Please do not make assertions in hook function.

Comment (crazycs520) [reply ddl/db_test.go]: Great

**Pull request merged**